OpenRA.Mods.Common/Activities/DeliverResources.cs

```diff
@@ -46,6 +46,9 @@ public override bool Tick(Actor self)
 			if (IsCanceling)
 				return true;
 
+			if (harv.IsTraitDisabled)
+				Cancel(self, true);
```

Comment:
I haven't tested this, but was expecting these to be above the `IsCanceling` check so that it could exit on the same tick. Does having this here work as expected?